Z-MirrorMove and Z-Copycat, ZMove Category

VeteranPadgett opened the pull request:
These moves Z-ify whatever attack they copy. I made changes for Me First too, but when it comes to executing the attack nothing happens. Also fixed ZAttacks to copy the physical/special category from the base move.

coyotte508 commented on src/BattleServer/moves.cpp:
```diff
@@ -4424,13 +4439,33 @@ struct MMMeFirst : public MM
+        } else {
+            MoveEffect::setup(move, s, t, b);
+            if (b.gen() >= 5) { // gen 3+4 done inline in calculateDamage
+                b.chainBp(s, 0x1800);
```
Isn't `clearBp() ` called in `useAttack()` anyway? MeFirst should add a turn function for `"BasePowerModifier"` after all the calls to `removeFunction`, and remove the `chainBp` call here.

VeteranPadgett commented:
You were right the chainbp call did nothing. Fixed that and moved the repetitive code into a function. I wasn't sure how to define it in one of the structures and be known by the other structures, so i put the function in MoveEffect.

Me First wasn't working before because the zmove checks use power == 0 to determine if it is an attack or status move, but Me First has a power of 1 which is changed to 0 mid attack using MoveSettings.

coyotte508 commented:
awesome! Well, defining it in the first of the three structures (`MMCopycat`) and calling `MMCopycat::reseMove` in the two ithers would have worked. But it being more general is fine too!
